internal/report/create.go

```diff
@@ -0,0 +1,152 @@
+// doesn't exist, setting the number of skipped policies in the
+// PolicyReport{}.Summary.
+func New(namespace *v1.ObjectReference, skippedNum int) error {
+	if policyReports[namespace.Name] == nil {
+		// policy report doesn't exist and needs to be created
+		policyReports[namespace.Name] = &PolicyReport{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              "polr-ns-" + namespace.Name,
+				Namespace:         namespace.Name,
+				CreationTimestamp: metav1.Now(),
+			},
+			Scope:  namespace, // namespace reference
+			Source: PolicyReportSource,
+			Summary: v1beta1.PolicyReportSummary{
+				Pass:  0,          // count of policies with requirements met
+				Fail:  0,          // count of policies with requirements not met
+				Warn:  0,          // not used for now
+				Error: 0,          // count of policies that couldn't be evaluated
+				Skip:  skippedNum, // count of policies that were not selected for evaluation
+			},
+			Results: []*v1beta1.PolicyReportResult{},
+		}
+	}
+	return nil
+}
+
+func Print() {
+	marshaled, err := json.Marshal(clusterPolicyReport)
+	if err != nil {
+		log.Error().Err(err).Msg("marshalling error")
+	}
+	fmt.Println(string(marshaled))
+	for _, report := range policyReports {
+		marshaled, err := json.Marshal(report)
+		if err != nil {
+			log.Error().Err(err).Msg("marshalling error")
+		}
+		fmt.Println(string(marshaled))
+	}
+}
+
+func AddResult(namespace string, policy policiesv1.Policy, resource unstructured.Unstructured, auditResponse *admv1.AdmissionReview, responseErr error) {
+	result := newResult(policy, resource, auditResponse, responseErr)
+	switch result.Result {
+	case "fail":
+		policyReports[namespace].Summary.Fail++
+	case "error":
+		policyReports[namespace].Summary.Error++
+	case "pass":
+		policyReports[namespace].Summary.Pass++
+	}
+	policyReports[namespace].Results = append(policyReports[namespace].Results, result)
+}
+
+func newResult(policy policiesv1.Policy, resource unstructured.Unstructured, auditResponse *admv1.AdmissionReview, responseErr error) *v1beta1.PolicyReportResult {
+	var result v1beta1.PolicyResult
+	var description string
+	if responseErr != nil {
+		result = "error"
+		description = auditResponse.Response.Result.Message
+	} else {
+		if auditResponse.Response.Allowed {
+			result = "pass"
+		} else {
+			result = "fail"
+			description = auditResponse.Response.Result.Message
+		}
+	}
+
+	var name string
+	switch policy.GetObjectKind().GroupVersionKind() {
+	case schema.GroupVersionKind{
+		Group:   constants.KubewardenPoliciesGroup,
+		Version: constants.KubewardenPoliciesVersion,
+		Kind:    "ClusterAdmissionPolicy",
+	}:
+		name = "cap-" + policy.GetName()
+	case schema.GroupVersionKind{
+		Group:   constants.KubewardenPoliciesGroup,
+		Version: constants.KubewardenPoliciesVersion,
+		Kind:    "AdmissionPolicy",
+	}:
+		name = "ap-" + policy.GetName()
+	}
+
+	time := metav1.Now()
+	timestamp := *time.ProtoTime()
+
+	log.Debug().Any("resource", resource)
```

Comment:
I've not run the code, does this log message provide any clue about the context where it's being originated?

Reply:
This was a leftover debug message, removed now.